motionplan/motionPlanner.go

```diff
@@ -400,11 +400,17 @@ func (mp *planner) getSolutions(ctx context.Context, seed referenceframe.FrameSy
 	ctxWithCancel, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	solutionGen := make(chan *ik.Solution, mp.planOpts.NumThreads*2)
+	solutionGen := make(chan *ik.Solution, mp.planOpts.NumThreads*20)
```

Comment:
Correct me if I'm wrong but this code does not actually fix the deadlock right?  It just makes it much less likely to happen (I may be reading this incorrectly).  If this is the case, would it not be better if we increase the channel size to something even larger? Is there a downside to doing this?

Reply:
You are correct; the addition of the `defer` is the fix for the deadlock. The intended result of the channel size increase is to unblock the IK solvers to ensure plenty of solutions are always available.

Increasing channel size does have the side effect of making the deadlock less likely but that's not why I'm doing it.